.teamcity/_self/projects/WebApp.kt

```diff
@@ -114,34 +122,42 @@ object BuildDockerImage : BuildType({
 			dockerPull = true
 			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
 		}
+		
+		val commonArgs = """
+			--label com.a8c.image-builder=teamcity
+			--label com.a8c.build-id=%teamcity.build.id%
+			--build-arg workers=32
+			--build-arg node_memory=32768
+			--build-arg use_cache=true
+			--build-arg base_image=%base_image%
+			--build-arg commit_sha=${Settings.WpCalypso.paramRefs.buildVcsNumber}
+			--build-arg manual_sentry_release=%MANUAL_SENTRY_RELEASE%
+			--build-arg is_default_branch=%teamcity.build.branch.is_default%
+			--build-arg sentry_auth_token=%SENTRY_AUTH_TOKEN%
+		""".trimIndent().replace("\n"," ")
 
-		dockerCommand {
+		script {
 			name = "Build docker image"
-			commandType = build {
-				source = file {
-					path = "Dockerfile"
-				}
-				namesAndTags = """
-					registry.a8c.com/calypso/app:build-%build.number%
-					registry.a8c.com/calypso/app:commit-${Settings.WpCalypso.paramRefs.buildVcsNumber}
-					registry.a8c.com/calypso/app:latest
-				""".trimIndent()
-				commandArgs = """
-					--pull
-					--label com.a8c.image-builder=teamcity
-					--label com.a8c.target=calypso-live
-					--label com.a8c.build-id=%teamcity.build.id%
-					--build-arg workers=32
-					--build-arg node_memory=32768
-					--build-arg use_cache=true
-					--build-arg base_image=%base_image%
-					--build-arg commit_sha=${Settings.WpCalypso.paramRefs.buildVcsNumber}
-					--build-arg manual_sentry_release=%MANUAL_SENTRY_RELEASE%
-					--build-arg is_default_branch=%teamcity.build.branch.is_default%
-					--build-arg sentry_auth_token=%SENTRY_AUTH_TOKEN%
-				""".trimIndent().replace("\n"," ")
-			}
-			param("dockerImage.platform", "linux")
+			scriptContent = """
+				#!/usr/bin/env bash
+				set -e
+				
+				docker build -f Dockerfile $commonArgs --pull \
+					--label com.a8c.target=calypso-live \
+					-t registry.a8c.com/calypso/app:build-%teamcity.build.number% \
+					-t registry.a8c.com/calypso/app:commit-${Settings.WpCalypso.paramRefs.buildVcsNumber} \
+					-t registry.a8c.com/calypso/app:latest \
+					. 2>&1 | tee docker_output.txt
+
+				cache_invalidated=false
+				if grep "resolving of build dependencies is invalid\|will re-resolve build dependencies\|invalidated because hashes differ" docker_output.txt ; then
```

Comment:
I added a few strings which can indicate when this happens -- I guess this is kind of hacky, because we're relying on weback to continue to use these exact strings. I think this is the easiest approach for now though.

Reply:
Plus, something else could technically output these strings... Anyways, I think this is something we can tweak moving forward